### PR TITLE
🎖️ Blog app - Views 🎖️

### PR DESCRIPTION
**Pull request by kambereBr**
## Project requirements

- ✅ No Linters error
- ✅ Used correct GitFlow.
- ✅ Used preprocessed HTML file with embedded Ruby code.
- ✅ Used layouts and templates for shared content.

### CODE REVIEW

File: app/views/partials/_user_profile.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    <p class='user_name'>
+     <%= link_to user_path (user) do %>
+      <%= user.name %><br>
+      <%end%>
+    </p>
+    <span class='u_details_post_number'>
+      Number of posts: <%= user.posts_counter%>
+    </span>
+  </div>
+</div>
```
Comment (Whoistolu) [inline app/views/partials/_user_profile.html.erb:16]: - You did a great job rendering partials 🥇 
- [ ] [REQUIRED] Kindly ensure that your design looks just like the wireframe provided. Styling is not required yet but it is a requirement that your design looks like the wireframe. Let's seeprate the div of each user image

Your Design
Screenshot 2023-11-04 at 15 30 25

Required Wireframe
Screenshot 2023-11-04 at 15 31 06

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,17 @@
...
+        <ul id='users_list'>
+            <% @user.three_recent_posts.each_with_index do |post, index| %>
+                <li>
+                    <%= render partial: "partials/post_item", locals: { post: post, index:index } %>
+                </li>
+            <% end %>
+        </ul>
+    </div>
+    <%= link_to 'See all posts', user_posts_path(@user.id), method: :get, class: 'btn btn-primary'%>
+</div>
```
Comment (Whoistolu) [inline app/views/users/show.html.erb]: - [ ] [REQUIRED] Well done job displaying the User show page, however, as I have recommended before, we need to ensure that the wireframe is designed just like the one provided in the project requirement. Let's adjust the design. you can reduce the width of the div and ensure the div for the user name aligns with the Bio div and the Posts div. Let's also not forget to separate the div for the image. lastly, do not forget to center the see all posy button and add the necessary border around it.

### Your Design
Screenshot 2023-11-04 at 15 36 48


### Required Wireframe
Screenshot 2023-11-04 at 15 37 15

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,19 @@
...
+          <%= render partial: "partials/comments_list", locals: { comments: comments} %>
+        </li>
+      <% end %>
+    </ul>
+  <%else%>
+    <%='No post'%>
+  <%end%>
+  </div>
+  <button id='btn_pagination'>Pagination</button>
+</div>
```
Comment (Whoistolu) [inline app/views/posts/index.html.erb]: - [ ] [REQUIRED] As recommended before now, let's adjust the design for the all post page to suite the recommended wireframe.

### Your Design
Screenshot 2023-11-04 at 15 46 33

### Required Wireframe
Screenshot 2023-11-04 at 15 48 47

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1 +1,13 @@
...
+      <span>Comment: <%=@post.comments_counter%>, </span>
+      <span>Like: <%=@post.likes_counter%> </span>
+   </div>
+   <div class="post_text">
+   <h3>
+     <%=@post.title%>
+   </h3>
+    <%= @post.text %>
+  </div>
+  <%= render partial: "partials/comments_list", locals: { comments: @comments} %>
```
Comment (Whoistolu) [inline app/views/posts/show.html.erb:12]: - [ ] [REQUIRED] You did an awesome job on the User Post show page, however, as I have recommended before, you need to ensure that this page design matches the recommended wireframe design in the project requirement. Let's ensure that the wireframe style for each post and its comments matches the wireframe.

### Your Design
Screenshot 2023-11-04 at 15 52 49

### Required Wireframe
Screenshot 2023-11-04 at 15 53 19

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @kambereBr 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlight
🔆 Users controller `#index` action renders all users to it's corresponding Views index page
🔆 Clicking on each user takes you to their show page
🔆 Uses `recent_post` method to render the user's 3 most recent post
🔆 See all Post** links opens user's post index
🔆 Uses `post.recent_comments` to render 5 most recent comments

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 4 above)
Comment (kambereBr) [reply app/views/posts/show.html.erb:12]: Dear @Whoistolu thanks for your review 👏

File: app/views/partials/_user_profile.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/partials/_user_profile.html.erb:16]: well resolved! ✔️

File: app/views/users/show.html.erb (hunk 2 above)
Comment (rloterh) [reply app/views/users/show.html.erb]: well resolved! ✔️

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (rloterh) [reply app/views/posts/index.html.erb]: well resolved! ✔️

File: app/views/posts/show.html.erb (hunk 4 above)
Comment (rloterh) [reply app/views/posts/show.html.erb:12]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Bruno,  

Your project is complete! 🏁 Great job for nailing it right on the first attempt 🥇There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to the view designs!
🎲 ReadMe looks great!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kambereBr** [comment]
Thanks dear @rloterh for your review

**Pull request merged**